experiments/benchmarks/benchmark_sim.jl

```diff
@@ -0,0 +1,169 @@
+using Test
+using Statistics
+import Profile, ProfileCanvas
+using Dates
+import ClimaLand
+using ClimaComms
+using CSV
+
+function parse_commandline()
+    s = ArgParseSettings()
+    @add_arg_table s begin
+        "--profiler"
+        help = "Profiler option: nsight or flamegraph"
+        arg_type = String
+        default = "integrated"
+    end
+    return parse_args(s)
+end
+
+function run_benchmarks(
+    device,
+    setup_simulation,
+    profiler,
+    PREVIOUS_GPU_TIME,
+    outdir,
+)
+    # run once just to ensure everything is compiled
+    simulation = setup_simulation()
+    ClimaLand.Simulations.solve!(simulation)
+    if profiler == "integrated"
+        @info "Running timing benchmarks"
+        (average_timing_s, std_timing_s) =
+            run_timing_benchmarks(device, setup_simulation)
+        @info "Done with timing benchmarks. Starting integrated profiling"
+        run_integrated_profiler(device, setup_simulation, outdir)
+        if get(ENV, "BUILDKITE_PIPELINE_SLUG", nothing) ==
+           "climaland-benchmark" && device isa ClimaComms.CUDADevice
+            if average_timing_s > PREVIOUS_GPU_TIME + std_timing_s
+                @info "Possible performance regression, previous average time was $(PREVIOUS_GPU_TIME)"
+            elseif average_timing_s < PREVIOUS_GPU_TIME - std_timing_s
+                @info "Possible significant performance improvement, please update PREVIOUS_GPU_TIME in $(@__DIR__)"
+            end
+            @testset "Performance" begin
+                @test PREVIOUS_GPU_TIME - 2std_timing_s <=
+                      average_timing_s <=
+                      PREVIOUS_GPU_TIME + 2std_timing_s
+            end
+        end
+    elseif profiler == "nsight"
+        run_external_profiler(setup_simulation)
+    else
+        @error "Unknown profiler: $profiler"
+    end
+    return
+
+end
+
+# integrated profiling for cpu - flame graphs are hard to interpret when gpu is used
+function run_integrated_profiler(device, setup_simulation, outdir)
+    simulation = setup_simulation()
+    Profile.@profile ClimaLand.Simulations.solve!(simulation)
+    results = Profile.fetch()
+    flame_file = joinpath(outdir, "flame_$device_suffix.html")
+    ProfileCanvas.html_file(flame_file, results)
+    @info "Saved compute flame to $flame_file"
+
+    simulation = setup_simulation()
+    Profile.Allocs.@profile sample_rate = 0.0025 ClimaLand.Simulations.solve!(
+        simulation,
+    )
+    results = Profile.Allocs.fetch()
+    profile = ProfileCanvas.view_allocs(results)
+    alloc_flame_file = joinpath(outdir, "alloc_flame_$device_suffix.html")
+    ProfileCanvas.html_file(alloc_flame_file, profile)
+    @info "Saved allocation flame to $alloc_flame_file"
+    return
+end
+
+# integrated profiling for CUDA
+function run_integrated_profiler(
+    device::ClimaComms.CUDADevice,
+    setup_simulation,
+    outdir,
+)
+    simulation = setup_simulation()
+    p = CUDA.@profile begin
+        for i in 1:20
```

Comment:
we run this profiler for 20 steps, but the other external option for 3. should these be the same?